File: sample/src/main/java/com/afollestad/aestheticsample/MainFragment.java
package com.afollestad.aestheticsample;

import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.Snackbar;
import android.support.v4.app.Fragment;
import android.support.v7.app.AlertDialog;
import android.support.v7.widget.SwitchCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.FrameLayout;
import android.widget.Spinner;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;
import com.afollestad.aesthetic.Aesthetic;
import com.afollestad.aesthetic.BottomNavBgMode;
import com.afollestad.aesthetic.BottomNavIconTextMode;
import io.reactivex.annotations.NonNull;
import io.reactivex.disposables.Disposable;
import io.reactivex.functions.Consumer;

/** @author Aidan Follestad (afollestad) */
public class MainFragment extends Fragment {

  @BindView(R.id.root)
  FrameLayout rootView;

  @BindView(R.id.switch_theme)
  SwitchCompat switchThemeView;

  @BindView(R.id.spinner)
  Spinner spinnerView;

  private Snackbar snackbar;
  private Unbinder unbinder;
  private Disposable isDarkSubscription;

  @Nullable
  @Override
  public View onCreateView(
      LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
    return inflater.inflate(R.layout.fragment_main, container, false);
  }

  @Override
  public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
    super.onViewCreated(view, savedInstanceState);
    unbinder = ButterKnife.bind(this, view);

    // Update the dark theme switch to the last saved isDark value.
    isDarkSubscription =
        Aesthetic.get()
            .isDark()
            .subscribe(
                new Consumer<Boolean>() {
                  @Override
                  public void accept(@NonNull Boolean isDark) {
                    switchThemeView.setChecked(isDark);
                  }
                });

    // Further view setup
    ArrayAdapter<String> spinnerAdapter =
        new ArrayAdapter<>(
            getContext(),
            R.layout.list_item_spinner,
            new String[] {
                "Spinner One",
                "Spinner Two",
                "Spinner Three",
                "Spinner Four",
                "Spinner Five",
                "Spinner Six"
            });
    spinnerAdapter.setDropDownViewResource(R.layout.list_item_spinner_dropdown);
    spinnerView.setAdapter(spinnerAdapter);
  }

  @Override
  public void onDestroyView() {
    isDarkSubscription.dispose();
    unbinder.unbind();
    super.onDestroyView();
  }

  @OnClick(R.id.switch_theme)
  public void onThemeChange(SwitchCompat switchCompat) {
    if (switchCompat.isChecked()) {
      Aesthetic.get()
          .activityTheme(R.style.AppThemeDark)
          .isDark(true)
          .textColorPrimaryRes(R.color.text_color_primary_dark)
          .textColorSecondaryRes(R.color.text_color_secondary_dark)
          .apply();
    } else {
      Aesthetic.get()
          .activityTheme(R.style.AppTheme)
          .isDark(false)
          .textColorPrimaryRes(R.color.text_color_primary)
          .textColorSecondaryRes(R.color.text_color_secondary)
          .apply();
    }
  }

  @OnClick(R.id.btn_dialog)
  public void showMaterialDialog() {
    new AlertDialog.Builder(getActivity())
        .setTitle(R.string.hello_world)
        .setMessage(R.string.lorem_ipsum)
        .setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() {
          @Override public void onClick(DialogInterface dialogInterface, int i) {

          }
        })
        .setNegativeButton(android.R.string.cancel, new DialogInterface.OnClickListener() {
          @Override public void onClick(DialogInterface dialogInterface, int i) {

          }
        })
        .show();
  }

  @OnClick({
      R.id.btn_black,
      R.id.btn_red,
      R.id.btn_purple,
      R.id.btn_blue,
      R.id.btn_green,
      R.id.btn_white
  })
  public void onClickButton(View view) {
    switch (view.getId()) {
      case R.id.btn_black:
        Aesthetic.get()
            .colorPrimaryRes(R.color.text_color_primary)
            .colorAccentRes(R.color.md_purple)
            .colorStatusBarAuto()
            .colorNavigationBarAuto()
            .bottomNavigationBackgroundMode(BottomNavBgMode.PRIMARY_DARK)
            .bottomNavigationIconTextMode(BottomNavIconTextMode.BLACK_WHITE_AUTO)
            .apply();
        break;
      case R.id.btn_red:
        Aesthetic.get()
            .colorPrimaryRes(R.color.md_red)
            .colorAccentRes(R.color.md_amber)
            .colorStatusBarAuto()
            .colorNavigationBarAuto()
            .bottomNavigationBackgroundMode(BottomNavBgMode.PRIMARY_DARK)
            .bottomNavigationIconTextMode(BottomNavIconTextMode.BLACK_WHITE_AUTO)
            .apply();
        break;
      case R.id.btn_purple:
        Aesthetic.get()
            .colorPrimaryRes(R.color.md_purple)
            .colorAccentRes(R.color.md_lime)
            .colorStatusBarAuto()
            .colorNavigationBarAuto()
            .bottomNavigationBackgroundMode(BottomNavBgMode.PRIMARY_DARK)
            .bottomNavigationIconTextMode(BottomNavIconTextMode.BLACK_WHITE_AUTO)
            .apply();
        break;
      case R.id.btn_blue:
        Aesthetic.get()
            .colorPrimaryRes(R.color.md_blue)
            .colorAccentRes(R.color.md_pink)
            .colorStatusBarAuto()
            .colorNavigationBarAuto()
            .bottomNavigationBackgroundMode(BottomNavBgMode.PRIMARY_DARK)
            .bottomNavigationIconTextMode(BottomNavIconTextMode.BLACK_WHITE_AUTO)
            .apply();
        break;
      case R.id.btn_green:
        Aesthetic.get()
            .colorPrimaryRes(R.color.md_green)
            .colorAccentRes(R.color.md_blue_grey)
            .colorStatusBarAuto()
            .colorNavigationBarAuto()
            .bottomNavigationBackgroundMode(BottomNavBgMode.PRIMARY_DARK)
            .bottomNavigationIconTextMode(BottomNavIconTextMode.BLACK_WHITE_AUTO)
            .apply();
        break;
      case R.id.btn_white:
        Aesthetic.get()
            .colorPrimaryRes(R.color.md_white)
            .colorAccentRes(R.color.md_blue)
            .colorStatusBarAuto()
            .colorNavigationBarAuto()
            .bottomNavigationBackgroundMode(BottomNavBgMode.PRIMARY)
            .bottomNavigationIconTextMode(BottomNavIconTextMode.SELECTED_ACCENT)
            .apply();
        break;
    }
  }

  @OnClick(R.id.fab)
  public void onClickFab() {
    if (snackbar != null) {
      snackbar.dismiss();
    }
    snackbar = Snackbar.make(rootView, R.string.hello_world, Snackbar.LENGTH_LONG);
    snackbar.setAction(
        android.R.string.cancel,
        new View.OnClickListener() {
          @Override
          public void onClick(View view) {
            // no-op
          }
        });
    snackbar.show();
  }
}

File: sample/src/main/java/com/afollestad/aestheticsample/MainFragment.kt
/*
 * Licensed under Apache-2.0
 *
 * Designed and developed by Aidan Follestad (@afollestad)
 */
package com.afollestad.aestheticsample

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.Fragment
import com.afollestad.aesthetic.Aesthetic
import com.afollestad.aesthetic.BottomNavBgMode
import com.afollestad.aesthetic.BottomNavIconTextMode
import com.google.android.material.snackbar.Snackbar
import io.reactivex.disposables.Disposable
import kotlinx.android.synthetic.main.fragment_main.btn_black
import kotlinx.android.synthetic.main.fragment_main.btn_blue
import kotlinx.android.synthetic.main.fragment_main.btn_green
import kotlinx.android.synthetic.main.fragment_main.btn_purple
import kotlinx.android.synthetic.main.fragment_main.btn_red
import kotlinx.android.synthetic.main.fragment_main.btn_white
import kotlinx.android.synthetic.main.fragment_main.view.btn_dialog
import kotlinx.android.synthetic.main.fragment_main.view.fab
import kotlinx.android.synthetic.main.fragment_main.view.spinner
import kotlinx.android.synthetic.main.fragment_main.view.switch_theme

/** @author Aidan Follestad (afollestad) */
class MainFragment : Fragment() {

  private var snackbar: Snackbar? = null
  private var isDarkSubscription: Disposable? = null

  override fun onCreateView(
    inflater: LayoutInflater,
    container: ViewGroup?,
    savedInstanceState: Bundle?
  ): View? {
    return inflater.inflate(R.layout.fragment_main, container, false)
  }

  override fun onViewCreated(
    view: View,
    savedInstanceState: Bundle?
  ) {
    super.onViewCreated(view, savedInstanceState)

    // Update the dark theme switch to the last saved isDark value.
    isDarkSubscription = Aesthetic.get()
        .isDark.subscribe { view.switch_theme.isChecked = it }

    // Further view setup
    val spinnerAdapter = ArrayAdapter(
        context!!,
        R.layout.list_item_spinner,
        arrayOf(
            "Spinner One", "Spinner Two", "Spinner Three",
            "Spinner Four", "Spinner Five", "Spinner Six"
        )
    )
    spinnerAdapter.setDropDownViewResource(R.layout.list_item_spinner_dropdown)
    view.spinner.adapter = spinnerAdapter

    view.switch_theme.setOnClickListener {
      if (view.switch_theme.isChecked) {
        Aesthetic.config {
          activityTheme(R.style.AppThemeDark)
          isDark(true)
          textColorPrimaryRes(R.color.text_color_primary_dark)
          textColorSecondaryRes(R.color.text_color_secondary_dark)
        }
      } else {
        Aesthetic.config {
          activityTheme(R.style.AppTheme)
          isDark(false)
          textColorPrimaryRes(R.color.text_color_primary)
          textColorSecondaryRes(R.color.text_color_secondary)
        }
      }
    }

    view.btn_dialog.setOnClickListener {
      AlertDialog.Builder(activity!!)
          .setTitle(R.string.hello_world)
          .setMessage(R.string.lorem_ipsum)
          .setPositiveButton(android.R.string.ok) { _, _ -> }
          .setNegativeButton(android.R.string.cancel) { _, _ -> }
          .show()
    }

    view.fab.setOnClickListener {
      snackbar?.dismiss()
      snackbar = Snackbar.make(view, R.string.hello_world, Snackbar.LENGTH_LONG)
      with(snackbar!!) {
        setAction(android.R.string.cancel) { _ -> }
        show()
      }
    }

    btn_black.setOnClickListener { onClickButton(it) }
    btn_red.setOnClickListener { onClickButton(it) }
    btn_purple.setOnClickListener { onClickButton(it) }
    btn_blue.setOnClickListener { onClickButton(it) }
    btn_green.setOnClickListener { onClickButton(it) }
    btn_white.setOnClickListener { onClickButton(it) }
  }

  override fun onDestroyView() {
    isDarkSubscription!!.dispose()
    super.onDestroyView()
  }

  private fun onClickButton(view: View) {
    when (view.id) {
      R.id.btn_black -> Aesthetic.config {
        colorPrimaryRes(R.color.text_color_primary)
        colorAccentRes(R.color.md_purple)
        colorStatusBarAuto()
        colorNavigationBarAuto()
        bottomNavigationBackgroundMode(BottomNavBgMode.PRIMARY_DARK)
        bottomNavigationIconTextMode(BottomNavIconTextMode.BLACK_WHITE_AUTO)
        swipeRefreshLayoutColorsRes(R.color.md_purple)
        attributeRes(R.attr.my_custom_attr, R.color.md_amber)
      }
      R.id.btn_red -> Aesthetic.config {
        colorPrimaryRes(R.color.md_red)
        colorAccentRes(R.color.md_amber)
        colorStatusBarAuto()
        colorNavigationBarAuto()
        bottomNavigationBackgroundMode(BottomNavBgMode.PRIMARY_DARK)
        bottomNavigationIconTextMode(BottomNavIconTextMode.BLACK_WHITE_AUTO)
        swipeRefreshLayoutColorsRes(R.color.md_red, R.color.md_amber)
        attributeRes(R.attr.my_custom_attr, R.color.md_blue)
      }
      R.id.btn_purple -> Aesthetic.config {
        colorPrimaryRes(R.color.md_purple)
        colorAccentRes(R.color.md_lime)
        colorStatusBarAuto()
        colorNavigationBarAuto()
        bottomNavigationBackgroundMode(BottomNavBgMode.PRIMARY_DARK)
        bottomNavigationIconTextMode(BottomNavIconTextMode.BLACK_WHITE_AUTO)
        swipeRefreshLayoutColorsRes(R.color.md_purple, R.color.md_lime)
        attributeRes(R.attr.my_custom_attr, R.color.md_green)
      }
      R.id.btn_blue -> Aesthetic.config {
        colorPrimaryRes(R.color.md_blue)
        colorAccentRes(R.color.md_pink)
        colorStatusBarAuto()
        colorNavigationBarAuto()
        bottomNavigationBackgroundMode(BottomNavBgMode.PRIMARY_DARK)
        bottomNavigationIconTextMode(BottomNavIconTextMode.BLACK_WHITE_AUTO)
        swipeRefreshLayoutColorsRes(R.color.md_blue, R.color.md_pink)
        attributeRes(R.attr.my_custom_attr, R.color.md_purple)
      }
      R.id.btn_green -> Aesthetic.config {
        colorPrimaryRes(R.color.md_green)
        colorAccentRes(R.color.md_blue_grey)
        colorStatusBarAuto()
        colorNavigationBarAuto()
        bottomNavigationBackgroundMode(BottomNavBgMode.PRIMARY_DARK)
        bottomNavigationIconTextMode(BottomNavIconTextMode.BLACK_WHITE_AUTO)
        swipeRefreshLayoutColorsRes(R.color.md_green, R.color.md_blue_grey)
        attributeRes(R.attr.my_custom_attr, R.color.md_pink)
      }
      R.id.btn_white -> Aesthetic.config {
        colorPrimaryRes(R.color.md_white)
        colorAccentRes(R.color.md_blue)
        colorStatusBarAuto()
        colorNavigationBarAuto()
        bottomNavigationBackgroundMode(BottomNavBgMode.PRIMARY)
        bottomNavigationIconTextMode(BottomNavIconTextMode.SELECTED_ACCENT)
        swipeRefreshLayoutColorsRes(R.color.md_blue)
        attributeRes(R.attr.my_custom_attr, R.color.md_lime)
      }
    }
  }
}
